includes/Services/AI_Service.php

```diff
@@ -0,0 +1,231 @@
+
+use function WordPress\AI\get_preferred_models;
+
+/**
+ * AI Service class.
+ *
+ * Manages AI provider configuration and provides a consistent interface
+ * for experimental features to communicate with AI providers.
+ *
+ * @since 0.1.0
+ */
+class AI_Service {
+
+	/**
+	 * Singleton instance.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @var \WordPress\AI\Services\AI_Service|null
+	 */
+	private static ?self $instance = null;
+
+	/**
+	 * Whether the service has been initialized.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @var bool
+	 */
+	private bool $initialized = false;
+
+	/**
+	 * Option key mapping from WordPress snake_case to SDK camelCase.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @var array<string, string>
+	 */
+	private static array $option_key_map = array(
+		'system_instruction' => 'systemInstruction',
+		'candidate_count'    => 'candidateCount',
+		'max_tokens'         => 'maxTokens',
+		'temperature'        => 'temperature',
+		'top_p'              => 'topP',
+		'top_k'              => 'topK',
+		'stop_sequences'     => 'stopSequences',
+		'presence_penalty'   => 'presencePenalty',
+		'frequency_penalty'  => 'frequencyPenalty',
+		'logprobs'           => 'logprobs',
+		'top_logprobs'       => 'topLogprobs',
+	);
+
+	/**
+	 * Gets the singleton instance.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @return \WordPress\AI\Services\AI_Service The singleton instance.
+	 */
+	public static function get_instance(): self {
+		if ( null === self::$instance ) {
+			self::$instance = new self();
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Private constructor to enforce singleton pattern.
+	 *
+	 * @since 0.1.0
+	 */
+	private function __construct() {}
+
+	/**
+	 * Initializes the AI service.
+	 *
+	 * This method should be called after AI_Client::init() on the WordPress 'init' hook.
+	 *
+	 * @since 0.1.0
+	 */
+	public function init(): void {
+		if ( $this->initialized ) {
+			return;
+		}
+
+		$this->initialized = true;
+
+		/**
+		 * Fires when the AI service is initialized.
+		 *
+		 * @since 0.1.0
+		 *
+		 * @param \WordPress\AI\Services\AI_Service $service The AI service instance.
+		 */
+		do_action( 'ai_service_initialized', $this );
+	}
+
+	/**
+	 * Checks if an AI provider is available and configured.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @return bool True if a provider is available, false otherwise.
+	 */
+	public function is_available(): bool {
+		/**
+		 * Filters whether the AI service is available.
+		 *
+		 * Allows developers or hosts to override availability detection,
+		 * for example when using pre-configured providers.
+		 *
+		 * @since 0.1.0
+		 *
+		 * @param bool|null $available Null to use default detection, or boolean to override.
+		 */
+		$available = apply_filters( 'ai_service_available', null );
+
+		if ( null !== $available ) {
+			return (bool) $available;
+		}
+
+		// Check if any provider credentials are configured.
+		$credentials = get_option( 'wp_ai_client_provider_credentials', array() );
```

Comment:
Consider using `API_Credentials_Manager::OPTION_PROVIDER_CREDENTIALS` instead of hard-coding this string.